lib/auth/auth.go

```diff
@@ -5081,6 +5125,73 @@ func (a *Server) submitAccessReview(
 		PromotedAccessListName: req.GetPromotedAccessListName(),
 	}
 
+	// Create a notification.
```

Comment:
the whole block seems to have a lot of repeated checks/code and I think can be abstracted. things to consider:
1. if the state is pending, we can just early return and not have to do any of these checks
2. since we've already created a pending check now, you only have to check for either approved or denied and then can assume its the other if it passes that check. this will remove a bit of indentation
3. the inner code after the state check seems to be pretty close/duplicated between approved denied so we could perhaps abstract this to one method and pass in the state as a param, and then return the appropriate struct from there. that would remove any extract check for approved/denied in this block anyway. for example `denied your access request for a resource.` and `approved your access request for a resource.` is only one word difference. the code you have no is fine btw, just a way to clean it up/make it more maintainable/readable.

Reply:
> if the state is pending, we can just early return and not have to do any of these checks

I might be mistaken, but I don't think it's possible for the state to ever be pending here. Since this is run when a reviewer reviews an access request, and it's only possible to approve or deny it

I've cleaned up the code as you suggested though, let me know what you think

Reply:
A request can still be pending if someone reviews it because it may require more than one approval/deny to get out of pending state (or i think 1 deny always denies but approval may need 2 or 3 or whatever)

Reply:
That's pretty cool, I didn't know that. Good catch